src/option_learning.py

```diff
@@ -212,3 +219,235 @@ def update_segment_from_option_spec(self, segment: Segment,
                                   dtype=np.float32)
                 option = param_opt.ground([robby, block], params)
                 segment.set_option(option)
+
+
+class _LearnedNeuralParameterizedOption(ParameterizedOption):
+    """A parameterized option that "implements" an operator.
+
+    * The option objects correspond to the operator parameters.
+    * The option initiable corresponds to the operator preconditions.
+    * The option terminal corresponds to the operator effects.
+    * The option policy is implemented as a neural regressor that takes in the
+      states of the option objects and the input continuous parameters (see
+      below), all concatenated into a 1D vector, and outputs an action.
+
+    The continuous parameters of the option are the main thing to note: they
+    correspond to a desired change in state for a subset of the option objects.
+    The objects that are changing correspond to changing_parameters; all other
+    objects are assumed to have no change in their state.
+
+    Note that the option terminal, which corresponds to the operator effects,
+    already describes how we want the objects to change. But these effects only
+    characterize a *set* of desired state changes. For example, a Pick operator
+    with Holding(?x) in the add effects says that we want to end up in *some*
+    state where we are holding ?x, but there are in general an infinite number
+    of such states. The role of the continuous parameters is to identify one
+    specific state in this set.
+
+    The hope is that by sampling different continuous parameters, the option
```

Comment:
so the idea is that the sampler is responsible for figuring out which objects' features it wants to change, and how much it wants to change them by? will it be easy to learn samplers like that downstream? hopefully...

Reply:
have you done any tests on this? just curious

Reply:
yeah that's the idea. we have run this all with sampler learning (you can see ashay's results above) and it works sometimes, but not as well as with an oracle sampler, as you'd expect

Reply:
oh and maybe you were asking whether we've tested whether the diversity thing works out as we want it to. no we haven't yet -- we should do this soon. making the covers environment hand regions nontrivial would help.